accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/AccessioningService.java

```diff
@@ -27,18 +27,39 @@
 
 import java.util.List;
 
+/**
+ * Service for retrieval and modifications of object accessions.
+ *
+ * @param <MODEL> Type of the objects identified by the accessions
+ * @param <HASH> Type of the hash calculated based on the fields that uniquely identify an accessioned object
+ * @param <ACCESSION> Accession of the object that identify a particular model
+ */
 public interface AccessioningService<MODEL, HASH, ACCESSION> {
 
+    /**
+     * Find the accessions associated with a list of objects.
+     * Searches object's accession in the repository, and if it does not exist, new accession is generated and stored in repository
+     *
+     * @param messages List of messages to be accessioned
+     * @return Wrappers containing the objects and their associated accession and hash
+     * @throws AccessionCouldNotBeGeneratedException when accession could not be generated
+     */
     List<AccessionWrapper<MODEL, HASH, ACCESSION>> getOrCreate(List<? extends MODEL> messages)
             throws AccessionCouldNotBeGeneratedException;
 
+    /**
+     * Find the accessions associated with a list of objects
+     *
+     * @param accessionedObjects List of already accessioned objects
+     * @return Wrappers containing the objects and their associated accession and hash
+     */
     List<AccessionWrapper<MODEL, HASH, ACCESSION>> get(List<? extends MODEL> accessionedObjects);
 
     /**
      * Finds last version of provided accession with its possible data model representations.
      *
-     * @param accession
-     * @return
+     * @param accession Accession of the object that identify a particular model
```

Comment:
`Accession that identifies a particular object`?

Reply:
Should we tailor words with method specific or be consistent across?

Reply:
Be consistent as much as you can and be specific as much as needed.

Reply:
`Accession that identifies object`